rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/TbGetTelemetryNode.java

```diff
@@ -116,21 +107,32 @@ private List<ReadTsKvQuery> buildQueries(Interval interval, List<String> keys) {
                 interval.getEndTs() - interval.getStartTs();
 
         return keys.stream()
-                .map(key -> new BaseReadTsKvQuery(key, interval.getStartTs(), interval.getEndTs(), aggIntervalStep, limit, aggregation, getOrderBy()))
+                .map(key -> new BaseReadTsKvQuery(key, interval.getStartTs(), interval.getEndTs(), aggIntervalStep, limit, aggregation, orderBy))
                 .collect(Collectors.toList());
     }
 
-    private String getOrderBy() {
+    private String getOrderBy() throws TbNodeException {
         switch (fetchMode) {
             case TbGetTelemetryNodeConfiguration.FETCH_MODE_ALL:
-                return orderByFetchAll;
+                return getOrderByFetchAll();
             case TbGetTelemetryNodeConfiguration.FETCH_MODE_FIRST:
                 return ASC_ORDER;
             default:
                 return DESC_ORDER;
         }
     }
 
+    private String getOrderByFetchAll() throws TbNodeException {
+        String orderBy = config.getOrderBy();
+        if (ASC_ORDER.equals(orderBy) || DESC_ORDER.equals(orderBy)) {
+            return orderBy;
+        }
+        if (StringUtils.isBlank(orderBy)) {
+            return ASC_ORDER;
+        }
+        throw new TbNodeException("Invalid fetch order selected.", true);
+    }
```

Comment:
The order might be null or blank only for all nodes created before the order by was added to the config. Based on this, the ideal solution will be to add the upgrade script that will add order by if it's null. In addition, we can change the value in the config from type String to Direction from the SortOrder class.